app_test.go

```diff
@@ -681,6 +681,76 @@ func TestAppStop(t *testing.T) {
 	})
 }
 
+func TestValidateApp(t *testing.T) {
+	t.Run("do not run provides on graph validation", func(t *testing.T) {
+		type type1 struct{}
+		err := ValidateApp(
+			Provide(func() *type1 {
+				panic("always")
```

Comment:
Let's make this a t.Error.

```suggestion
				t.Error("provide must not be called")
				return nil
```